src/test/regress/sql/multi_level_recursive_queries.sql

```diff
@@ -0,0 +1,115 @@
+-- multi recursive queries with joins, subqueries, and ctes
+CREATE SCHEMA multi_recursive;
+SET search_path TO multi_recursive;
+
+
+DROP TABLE IF EXISTS tbl_dist1;
+CREATE TABLE tbl_dist1(id int);
+SELECT create_distributed_table('tbl_dist1','id');
+
+DROP TABLE IF EXISTS tbl_ref1;
+CREATE TABLE tbl_ref1(id int);
+SELECT create_reference_table('tbl_ref1');
+
+INSERT INTO tbl_dist1 SELECT i FROM generate_series(0,10) i;
+INSERT INTO tbl_ref1 SELECT i FROM generate_series(0,10) i;
+
+-- QUERY1
```

Comment:
These are very complex queries. Can we do 2 things:

1) Add the simple example as we have on the issue:
```SQL
SELECT t1.id
FROM (
    SELECT t2.id
    FROM (
        SELECT t0.id
        FROM tbl_dist1 t0
        LIMIT 5        
    ) AS t2
    INNER JOIN tbl_dist1 AS t3 USING (id)
) AS t1
FULL JOIN tbl_dist1 t4 USING (id);
```

2) per query here, can you add briefly what happens during the planning as a comment? For example, some queries work. How/why? And others fail, how/why?

Thanks